components/shared/embedder/input_events.rs

```diff
@@ -280,3 +284,62 @@ pub enum GamepadUpdateType {
     /// <https://www.w3.org/TR/gamepad/#dfn-represents-a-standard-gamepad-button>
     Button(usize, f64),
 }
+
+pub trait WebDriverInputEvent {
+    fn webdriver_id(&self) -> Option<WebDriverMessageId>;
+    fn with_webdriver_id(&mut self, webdriver_id: Option<WebDriverMessageId>) -> Self;
+}
+
+impl WebDriverInputEvent for InputEvent {
+    fn webdriver_id(&self) -> Option<WebDriverMessageId> {
+        match self {
+            InputEvent::EditingAction(..) => None,
+            InputEvent::Gamepad(..) => None,
+            InputEvent::Ime(..) => None,
+            InputEvent::Keyboard(..) => None,
+            InputEvent::MouseButton(event) => event.webdriver_id,
+            InputEvent::MouseMove(event) => event.webdriver_id,
+            InputEvent::Touch(..) => None,
+            InputEvent::Wheel(..) => None,
+        }
+    }
+
+    fn with_webdriver_id(&mut self, webdriver_id: Option<WebDriverMessageId>) -> Self {
```

Comment:
```suggestion
    fn with_webdriver_id(mut Self, webdriver_id: Option<WebDriverMessageId>) -> Self {
```
This way we get rid of the clone.

Reply:
I tried this but the compiler doesn't encourage us to use `mut self`.